<?xml version="1.0"?>
<launch>

    <param name="use_sim_time" value="false"/>
    <!-- Arguments -->
    <arg name="debug" default="false" />
    <arg unless="$(arg debug)" name="launch_prefix" value="" />
    <arg     if="$(arg debug)" name="launch_prefix" value="gdb --ex run --args" />

    <arg name="occmap" default="simple" />

    <!-- Parameters -->
    <rosparam file="$(find ompl_planners_ros)/param/rrtstar_motion_planner_params_stefmap.yaml" command="load"/>

    <!-- Nodes -->
    <node name="map_server" pkg="map_server" type="map_server" args="$(find ompl_planners_ros)/maps/$(arg occmap).yaml"/>

    <node name="stefmap_server" pkg="stefmap_ros" type="stefmap_server.py" />

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find ompl_planners_ros)/launch/srscp.rviz"/>

    <node name="RRTStarMotionPlanner" type="car_planner_ros_node" pkg="ompl_planners_ros" launch-prefix="$(arg launch_prefix)">
        <remap from="goal" to="move_base_simple/goal"/>
        <remap from="start" to="move_base_simple/start"/>
    </node>

</launch>
